lambda/update/parse/changes.go

```diff
@@ -92,19 +101,29 @@ func Validate(fn func() []shared.FieldError) Option {
 // Then to parse to a string s do:
 //
 //	parser.Field("/thing", &s)
-func (p *Parser) Field(key string, v any, opts ...Option) *Parser {
+func (p *Parser) Field(key string, existing any, opts ...Option) *Parser {
 	options := fieldOpts{}
 	for _, opt := range opts {
 		options = opt(options)
 	}
 
 	for i, change := range p.changes {
 		if change.Key == key {
-			if !bytes.Equal(change.Old, []byte("null")) {
-				p.errors = append(p.errors, shared.FieldError{Source: change.Source("/old"), Detail: "must be null"})
+			// Adding for expand phase - will be removed once consumers are updated
+			if options.mustMatchExisting {
+				var old any
+				if err := json.Unmarshal(change.Old, &old); err != nil {
```

Comment:
Since this works, it is fine. But did you try going in the other direction with comparing `change.Old` with `json.Marshal(existing)`? That was going to be my initial plan.

Reply:
Yeah I kept encountering problems with comparing pointers when and I ended up in reflection land so this felt preferable although happy to use that approach if I missed a trick somewhere?